boofuzz/sessions.py

```diff
@@ -1381,8 +1419,9 @@ def _post_send(self, target):
             except exception.BoofuzzTargetConnectionFailedError:
                 self._fuzz_data_logger.log_fail(constants.ERR_CONN_FAILED)
             except Exception:
-                self._fuzz_data_logger.log_fail(
-                    "Custom post_send method raised uncaught Exception." + traceback.format_exc())
+                self._fuzz_data_logger.log_error(constants.ERR_CALLBACK_FUNC.format(func_name="post_send")
+                                                 + traceback.format_exc())
+                raise exception.BoofuzzCallbackError()
```

Comment:
The three callbacks all use similar code. Could we factor it out into one function? E.g.:
```python
def _execute_callback_list(callbacks, name):
    ...

...

def _post_send(self, target):
    ...
    self._execute_callback_list(self._post_test_case_methods, 'post- test case')
```

Reply:
I thought about it but they do differ in some points like exception handling.
I will try to figure a way out.

Reply:
Looking at the code, the only part I see that we could factor out would be:
```python
for f in self._post_test_case_methods:
    self._fuzz_data_logger.open_test_step('{1} callback: "{0}"'.format(f.__name__), test_step_name)
    f(target=target, fuzz_data_logger=self._fuzz_data_logger, session=self, sock=target)
```
The exception handling takes up the largest part of those functions but we can't put that in a generic callback method. Not that I see a way to do so, at least.

If you like I can still put those 3 lines in an extra function.

Reply:
@SR4ven I think it would be good if more of the code were the same anyway -- e.g., it makes sense for `BoofuzzTargetConnectionReset` to be handled the same way no matter which kind of callback is in play... at least I think it makes sense.

That said, this code cleanliness recommendation can be tabled for another time. No need to work through it right now.